examples/tv-app/tv-common/clusters/messages/MessagesManager.h

```diff
@@ -21,6 +21,97 @@
 
 #include <iostream>
 #include <list>
+#include <vector>
+
+/**
+ * Note on memory management:
+ *
+ * The CachedMessage contains strings and objects
+ * allocated when created (via HandlePresentMessagesRequest) and freed
+ * in the destructor (via HandleCancelMessagesRequest).
+ *
+ */
+
+struct CachedMessageOption
+{
+    uint32_t mId;
+    std::string mLabel;
+    chip::app::Clusters::Messages::MessageResponseOption mOption;
+
+    CachedMessageOption(uint32_t id, std::string label) :
+        mId(id), mLabel(label),
+        mOption{ chip::Optional<uint32_t>(mId), chip::Optional<chip::CharSpan>(chip::CharSpan::fromCharString(mLabel.c_str())) }
```

Comment:
Might be more readable as:
```suggestion
        mOption{ chip::MakeOptional(id), chip::MakeOptional(chip::CharSpan::fromCharString(mLabel.c_str())) }
```

Reply:
addressed